instance/models/instance.py

```diff
@@ -0,0 +1,476 @@
+    ansible_playbook_name = models.CharField(max_length=50, default='edx_sandbox')
+    ansible_extra_settings = models.TextField(blank=True)
+
+    # List of attributes to include in the settings output
+    ANSIBLE_SETTINGS = ['ansible_extra_settings']
+
+    class Meta:
+        abstract = True
+
+    @property
+    def ansible_playbook_filename(self):
+        """
+        File name of the ansible playbook
+        """
+        return '{}.yml'.format(self.ansible_playbook_name)
+
+    @property
+    def inventory_str(self):
+        """
+        The ansible inventory (list of servers) as a string
+        """
+        inventory = ['[app]']
+        server_model = self.server_set.model
+        for server in self.server_set.filter(status=server_model.BOOTED).order_by('created'):
+            inventory.append(server.public_ip)
+        inventory_str = '\n'.join(inventory)
+        self.log('debug', 'Inventory for instance {}:\n{}'.format(self, inventory_str))
+        return inventory_str
+
+    @property
+    def vars_str(self):
+        """
+        The ansible vars (private configuration) as a string
+        """
+        template = loader.get_template('instance/ansible/vars.yml')
+        vars_str = template.render({'instance': self})
+        for attr_name in self.ANSIBLE_SETTINGS:
+            additional_vars = getattr(self, attr_name)
+            vars_str = ansible.yaml_merge(vars_str, additional_vars)
+        self.log('debug', 'Vars.yml for instance {}:\n{}'.format(self, vars_str))
+        return vars_str
+
+    def run_playbook(self, playbook_name=None):
+        """
+        Run a playbook against the instance active servers
+        """
+        with clone_configuration_repo() as configuration_repo_path:
+            playbook_path = os.path.join(configuration_repo_path, 'playbooks')
+            requirements_path = os.path.join(configuration_repo_path, 'requirements.txt')
+
+            self.log('info', 'Running playbook "{path}/{name}" for instance {instance}...'.format(
+                path=playbook_path,
+                name=playbook_name,
+                instance=self,
+            ))
+
+            log_lines = []
+            with ansible.run_playbook(
+                requirements_path,
+                self.inventory_str,
+                self.vars_str,
+                playbook_path,
+                self.ansible_playbook_filename,
+                username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
+            ) as processus:
+                for line in processus.stdout:
+                    line = line.decode('utf-8').rstrip()
+                    self.log('info', line)
+                    log_lines.append([line.rstrip()])
+
+        self.log('info', 'Playbook run completed for instance {}'.format(self))
+        return log_lines
+
+
+# Open edX ####################################################################
+
+class OpenEdXInstanceQuerySet(models.QuerySet):
```

Comment:
This class is not really specific to OpenEdX. I think you should make `DEFAULT_FORK` an attribute/property of the model, and then rename this so it is a generic manager/queryset that can be used for any type of instance.